### PR TITLE
[istio] add sampling param for tracing

### PR DESCRIPTION
**Pull request by fl64**
## Description
Added global parameter for configuring tracing sample rate.

## Why do we need it, and what problem does it solve?

Closes https://github.com/deckhouse/deckhouse/issues/2136

## What is the expected result?

Now you can configure tracing sample rate value via deckhouse config.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: istio
type: feature
summary: Added global parameter for configuring tracing sample rate.
impact_level: default
```

### CODE REVIEW

File: ee/modules/110-istio/openapi/doc-ru-config-values.yaml [hunk 1]
```diff
@@ -71,6 +71,8 @@ properties:
     properties:
       enabled:
         description: Включить отправку трейсов в коллектор и интегрировать их отображение в kiali.
+      sampling:
+        description: Значение частоты дискретизации. Частота дискретизации должна находиться в диапазоне от 0,0 до 100,0 с точностью до 0,01.
```
Comment (apolovov) [inline ee/modules/110-istio/openapi/doc-ru-config-values.yaml:75]:
```suggestion
        description: Значение частоты дискретизации. Частота дискретизации должна находиться в диапазоне от 0.0 до 100.0 с точностью до 0.01.
```

**Review by apolovov — DISMISSED** [review]

File: ee/modules/110-istio/openapi/config-values.yaml [hunk 2]
```diff
@@ -138,6 +138,14 @@ properties:
         description: Turn on or off tracing collection and displaying in kiali.
         default: false
         x-examples: [true]
+      sampling:
```
Comment (nabokihms) [inline ee/modules/110-istio/openapi/config-values.yaml:141]: Could you please elaborate more on this option and why tuning this is required?

File: ee/modules/110-istio/openapi/config-values.yaml [hunk 3]
```diff
@@ -138,6 +138,14 @@ properties:
...
+        type: number
+        minimum: 0.01
+        maximum: 100.0
```
Comment (nabokihms) [inline ee/modules/110-istio/openapi/config-values.yaml:144]: From this description, it is hard to tell units. Percents?

Comment (fl64) [reply ee/modules/110-istio/openapi/config-values.yaml:144]: Yes, it's a percentage. Add units to the description of config-values?

File: ee/modules/110-istio/openapi/config-values.yaml [hunk 4]
```diff
@@ -138,6 +138,14 @@ properties:
...
+        multipleOf: 0.01
+        description: Value of sampling rate in percents. The sampling rate should be in the range of 0.0 to 100.0 with a precision of 0.01.
```
Comment (apolovov) [inline ee/modules/110-istio/openapi/config-values.yaml:146]:
```suggestion
        description: |
          Sampling rate in percents. The rate should be in the range of 0.0 to 100.0 with a precision of 0.01. It is possible to override this setting with Pod annotation:
          ```
          proxy.istio.io/config: |
            tracing:
              sampling: 100.0
          ```
```

File: ee/modules/110-istio/openapi/doc-ru-config-values.yaml [hunk 5]
```diff
@@ -71,6 +71,8 @@ properties:
     properties:
       enabled:
         description: Включить отправку трейсов в коллектор и интегрировать их отображение в kiali.
+      sampling:
+        description: Значение частоты дискретизации в процентах. Частота дискретизации должна находиться в диапазоне от 0.0 до 100.0 с точностью до 0.01.
```
Comment (apolovov) [inline ee/modules/110-istio/openapi/doc-ru-config-values.yaml:75]:
```suggestion
        description: |
          Значение процентной доли запросов, которые будут сэмплированы для отправки в jaeger. Значение должно находиться в диапазоне от 0.0 до 100.0 с точностью до 0.01. Значение можно переопределить для каждого пода индивидуально при помощи аннотации:
          ```
          proxy.istio.io/config: |
            tracing:
              sampling: 100.0
          ```
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @apolovov. I've started the workflow for label 'deploy/web/test'...

File: ee/modules/110-istio/openapi/config-values.yaml [hunk 6]
```diff
@@ -138,6 +138,20 @@ properties:
...
         default: false
         x-examples: [true]
+      sampling:
+        type: number
+        minimum: 0.01
+        maximum: 100.0
+        multipleOf: 0.01
+        description: |
+          Sampling rate in percents. The rate should be in the range of 0.0 to 100.0 with a precision of 0.01. It is possible to override this setting with Pod annotation:
+          ```
```
Comment (apolovov) [inline ee/modules/110-istio/openapi/config-values.yaml:148]:
```suggestion
          ```yaml
```

File: ee/modules/110-istio/openapi/config-values.yaml [hunk 7]
```diff
@@ -138,6 +138,20 @@ properties:
...
+          proxy.istio.io/config: |
+            tracing:
+              sampling: 100.0
+          ```
+        default: 1.0
+        x-examples: [1, 1.0, 1.01, 50.0, 100.0, 100]
```
Comment (apolovov) [inline ee/modules/110-istio/openapi/config-values.yaml:154]:
```suggestion
        x-examples: [50.05]
```

File: ee/modules/110-istio/openapi/doc-ru-config-values.yaml [hunk 8]
```diff
@@ -71,6 +71,14 @@ properties:
     properties:
       enabled:
         description: Включить отправку трейсов в коллектор и интегрировать их отображение в kiali.
+      sampling:
+        description: |
+          Процентная доля запросов, которые будут сэмплированы для отправки в jaeger.  Значение должно находиться в диапазоне от 0.0 до 100.0 с точностью до 0.01. Его можно переопределить для каждого пода индивидуально при помощи аннотации:
+          ```
```
Comment (apolovov) [inline ee/modules/110-istio/openapi/doc-ru-config-values.yaml:77]:
```suggestion
          ```yaml
```

**Review by apolovov — DISMISSED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @apolovov. I've started the workflow for label 'deploy/web/test'...

**Review by apolovov — APPROVED** [review]

File: ee/modules/110-istio/openapi/config-values.yaml (hunk 2 above)
Comment (nabokihms) [reply ee/modules/110-istio/openapi/config-values.yaml:141]: My question was more about the `sampling` term. May be we can provide a link to some documentation that can clarify what sampling is.

**Review by apolovov — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

**Pull request merged**